server/src/main/java/io/helidon/extensions/mcp/server/McpSamplingImageContent.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.extensions.mcp.server;
+
+import java.util.Base64;
+
+import io.helidon.common.media.type.MediaType;
+
+/**
+ * MCP sampling image content.
+ */
+public final class McpSamplingImageContent implements McpSamplingMessage, McpImageContent {
```

Comment:
Same as above - implementation classes should not be public.
If you need something in API that implements both `McpSamplingImage` and `McpImageContent`, create a new interface for it. Using a final class makes it hard to test, and is not extensible for the future.